### PR TITLE
Create recycler

### PR DESCRIPTION
**Pull request by coffeepac**
This works now:
- retries a message up to MAX_RETRY_TIMES (currently 5, will be a parameter eventually)
- delays a message it has seen before by MIN_RETRY_WAIT (currently 5 minutes, will be a parameter eventually)
- throws messages to its error pipeline if there are too many retries
- includes circleci bits for testing and deploying

### CODE REVIEW

**Comment by doomsuckle** [comment]
Might also want to add a "jitter" parameter so you can be like, min(5 sec, gaus(5 sec, 2 sec))

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+              <goal>shade</goal>
+            </goals>
+            <configuration>
+              <finalName>${project.artifactId}-bundled-${project.version}</finalName>
+              <artifactSet>
+                <includes>
+                  <include>*:*</include>
+                </includes>
+              </artifactSet>
+              <filters>
```
Comment (doomsuckle) [inline pom.xml:64]: XML is stupid

**Comment by coffeepac** [comment]
add jitter to what.  the test or the actual waiting?

**Comment by doomsuckle** [comment]
the actual

**Comment by coffeepac** [comment]
what does adding jitter to the actual code path get us?

**Comment by doomsuckle** [comment]
```
¯\_(ツ)_/¯ You raised the issue of a clumping effect. It solves that. ;)
```

**Comment by coffeepac** [comment]
ooooh, gotcha. that could break up the clumping.  I don't think the
clumping matters though, goog infra provides enough elasticity to not have
to worry about it from a perf standpoint.  I was more raising the issue as
a 'this is a possible emergent behavior of our system during times of
duress' thing.  I'll leave jitter out for now but we have this conversation
to get back to if we want it.

**Comment by doomsuckle** [comment]
+1 :)

**Pull request merged**